include/kafka/Consumer.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
```

Comment:
Not sure if this will confuse people when we have both a Consumer.h and KafkaConsumer.h?

Reply:
The `Consumer` is a namespace, currently, the file is for some datatypes. `KafkaConsumer` is only for the Kafka client. In future, there will be a dedicated PR to refine the namespace hierarchy. Generally , would follow https://kafka.apache.org/27/javadoc/org/apache/kafka/clients/consumer/package-summary.html  
Then, the file might be renamed as something likes `ConsumerCommon`, and `KafkaConsumer` would be in the namespace `Consumer` as well.

Reply:
ConsumerCommon makes better sense to me

Reply:
ConsumerCommon looks much a better name for me. Consumer.h is an interface in Java, where KafkaConsumer inherits from, so it will create less confusion.

In Java, I also see ConsumerGroupMetaData is separated into a dedicated file. If we separate ConsumerGroupMetaData into a dedicated file, it may also be fine to merge back the RebalanceEventType, RebalanceCallback and OffsetCommitCallback into KafkaConsumer.h

Reply:
Yeah, I agree. There would be another PR coming, which would includes many changes to refine the namespace hierarchy. E.g. `KafkaConsumer` might be in the namespace `Consumer` as well, and also considering whether it makes sense to create an outer most namespace `kafka::client`, for `Consumer`/`Producer`, ...